tests/README.rst

```diff
@@ -80,6 +80,125 @@ To do that, run `podman pull quay.io/rhinstaller/anaconda-ci:master` or build
 it locally again.
 
 
+Explanation of the GitHub configuration
+---------------------------------------
+
+Pull request for master:
+________________________
+
+Everything is stored in `.github/workflows/validate.yml`_ the workflow.
+These files completely describe what steps are required to run the tests, what
+are the triggers and so on. We use GitHub runners here so we don't have to care
+too much about what is executed there. GitHub runners are designed to be isolated
+and secure. On master we are building the container only if the container files
+have changed, otherwise we are pulling the container from our repository. For more
+information see below.
+
+Pull request for RHEL:
+______________________
+
+Again configuration is stored in `.github/workflows/validate-rhel-8.yml`_ the
+workflow. However, here we also have (fully automatically deployed) runners in
+our Upshift instance. GitHub configuration is using our self-hosted runners.
+These runners are ``anaconda-ci:rhel8`` containers with all the dependencies
+in place so the yml configuration will just execute tests. 
+You can start runners locally by running the container and
+providing GitHub token. That is pretty valuable in case of workflow testing.
+See `github-action_run_once <https://github.com/rhinstaller/anaconda/blob/rhel-8/dockerfile/anaconda-ci/github-action-run-once>`_
+for more details.
+
+To protect our self-hosted runners, tests only run automatically for
+`rhinstaller organization members <https://github.com/orgs/rhinstaller/people>`_.
+For external contributors, an organization member needs to approve the test run
+by sending a comment starting with ``/tests``.
+
+
+What role do the containers have
+--------------------------------
+
+All active branches run tests in containers. Containers have all the
+dependencies installed and the environment prepared to run tests or connect our
+GitHub runners (used by RHEL only).
+
+Automatic container build
+_________________________
+
+Containers are updated daily by `.github/workflows/container-autoupdate.yml`_
+GitHub action from Anaconda ``master`` repository. Before pushing a new
+container, tests are executed on this container to avoid regressions.
+
+Manual container build
+______________________
+
+There are two possible ways how to build containers manually.
```

Comment:
Now this section just defines one, so drop this sentence and the following "GitHub:"